### PR TITLE
feat: 🎸 ignore result of job runner if job has been canceled

### PR DESCRIPTION
**Pull request by severo**
also: refactor to remove two queue methods (kill_zombies, kill_long_jobs): job_manager is now in charge of finishing the jobs, and updating the cache (if needed).

### CODE REVIEW

**Review by albertvillanova — APPROVED** [review]
Thank you.

File: services/worker/tests/test_job_manager.py [hunk 1]
```diff
@@ -171,24 +192,24 @@ def test_job_runner_set_crashed(
     test_processing_step: ProcessingStep,
```
Comment (AndreaFrancis) [inline services/worker/tests/test_job_manager.py:192]: Not part of this PR, but I should have changed the name of this test, maybe only test_set_crashed since the logic is now handled by job_manager and not job_runner.

**Review by AndreaFrancis — APPROVED** [review]
Thanks! Nice refactor.

File: services/worker/tests/test_job_manager.py (hunk 1 above)
Comment (severo) [reply services/worker/tests/test_job_manager.py:192]: OK... Let's change this when we see it again in a PR

**Review by lhoestq — APPROVED** [review]
nice ! :)

**Comment by severo** [comment]
follow-up: https://github.com/huggingface/datasets-server/pull/1227
